.github/workflows/constraints.txt

```diff
@@ -1,3 +1,3 @@
-pip==24.0
-poetry==1.3.2
-virtualenv==20.21.0
+pip==24.3.1
+poetry==1.8.5
```

Comment:
It seems that poetry is compatible with python 3.12 from version 1.7.0 onwards. In the future, it will also be necessary to support 3.13, etc., so we are upgrading to the latest version.

- https://python-poetry.org/history/#170---2023-11-03

If we only upgraded poetry, it would have conflicted with the requirements of virtualenv, so we are also upgrading virtualenv to the latest version.